src/utils/getRoundedAmount.ts

```diff
@@ -3,5 +3,13 @@ export function getRoundedAmount(balance: string, fractionDigits: number) {
     return balance;
   }
   const parsedBalance = Number.parseFloat(balance);
-  return Number(parsedBalance)?.toFixed(fractionDigits).replace(/0+$/, '');
+  const result = Number(parsedBalance)
+    ?.toFixed(fractionDigits)
+    .replace(/0+$/, '');
+
+  if (parsedBalance > 0 && Number.parseFloat(result) === 0) {
```

Comment:
nit. add a line of comment explaining this case. I know it has a test, which is great. But have a line of comment helps remember what this part is covering for.